util/network-devp2p/src/host.rs

```diff
@@ -285,19 +287,30 @@ impl Host {
 			Some(addr) => addr,
 		};
 
+		let mut key_created = false;
 		let keys = if let Some(ref secret) = config.use_secret {
 			KeyPair::from_secret(secret.clone())?
 		} else {
-			config.config_path.clone().and_then(|ref p| load_key(Path::new(&p)))
+			config.config_path.clone().and_then(|ref p| load(Path::new(&p)))
 				.map_or_else(|| {
+				key_created = true;
 				let key = Random.generate();
 				if let Some(path) = config.config_path.clone() {
-					save_key(Path::new(&path), key.secret());
+					save(Path::new(&path), key.secret());
 				}
 				key
 			},
 			|s| KeyPair::from_secret(s).expect("Error creating node secret key"))
 		};
+		let mut enr = None;
+		if !key_created {
+			if let Some(path) = &config.config_path {
+				if let Some(data) = load(Path::new(&path)) {
+					enr = EnrManager::load(keys.secret().clone(), data);
+				}
+			}
+		}
+		let enr = enr.unwrap_or_else(|| EnrManager::new(keys.secret().clone(), 0).expect("Key is always valid; qed"));
```

Comment:
Could we elaborate on why `Key is always valid`? I imagine something like:
```
keys.secret() is a valid secp256k1 secret; Enr does not fail given valid secp256k1 secret; qed
```

Reply:
Done